hot-reload-analysis/src/main/kotlin/org/jetbrains/compose/reload/analysis/utils.kt

```diff
@@ -36,3 +36,11 @@ internal fun Checksum.updateString(value: String) {
     updateInt(value.length)
     update(value.encodeToByteArray())
 }
+
+internal fun <K, V> HashMap<K, V>.toImmutableHashMap(): Map<K, V> = when (size) {
```

Comment:
The name seems a little bit off 🤔 
It's not the primary purpose to be immutable (it returns a read-only view that can be mutated if we really want to). How about `toHashMap()` or `toReadOnlyHashMap()`